src/builtins/BuiltinTypedArray.cpp

```diff
@@ -485,6 +485,100 @@ static Value builtinTypedArrayCopyWithin(ExecutionState& state, Value thisValue,
     return O;
 }
 
+template <const bool isForward>
+static Value fastTypedArrayIndexSearch(TypedArrayObject* arr, size_t k, size_t len, const Value& value)
+{
+    auto type = arr->typedArrayType();
+    if (type == TypedArrayType::BigInt64 || type == TypedArrayType::BigUint64) {
+        if (!value.isBigInt()) {
+            return Value(-1);
+        }
+    } else {
+        if (!value.isNumber()) {
+            return Value(-1);
+        }
+    }
+
+    uint8_t* buffer = arr->rawBuffer();
+    auto elementSize = arr->elementSize();
+    int64_t byteK = static_cast<int64_t>(k * elementSize);
+    int64_t byteLength = static_cast<int64_t>(len * elementSize);
+
+    auto compFn = [](int64_t byteK, int64_t byteLength) -> bool {
+        if (isForward) {
+            return byteK < byteLength;
+        } else {
+            return byteK >= 0;
+        }
+    };
+
+    auto updateFn = [](int64_t& byteK, const int diff) {
+        if (isForward) {
+            byteK += diff;
+        } else {
+            byteK -= diff;
+        }
+    };
+
+    if (!isForward && byteK >= byteLength) {
+        return Value(-1);
+    }
+
+    if (type == TypedArrayType::BigInt64 || type == TypedArrayType::BigUint64) {
+        uint64_t argv0ToUint64 = type == TypedArrayType::BigInt64 ? bitwise_cast<uint64_t>(value.asBigInt()->toInt64()) : value.asBigInt()->toUint64();
+        while (compFn(byteK, byteLength)) {
+            if (*reinterpret_cast<uint64_t*>(&buffer[byteK]) == argv0ToUint64) {
```

Comment:
Is it fine to directly access array elements regardless of Endian?
What if we use BigEndian?

Reply:
IMO We don't need to care endianness.
because two values are comes from same machine